app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -52,98 +70,34 @@ private function parseReferralCode($referralCode)
-                        break;
-
-                    case 'campaign':
-                        $values['campaign_id'] = $value[1];
-                        break;
-
-                    case 'campaignrunid':
-                        $values['campaign_run_id'] = $value[1];
-                        break;
-
-                    case 'source':
-                        $values['source'] = $value[1];
-                        break;
-
-                    case 'source_details':
-                        $values['source_details'] = $value[1];
-                        break;
+        if (empty($referralCode)) {
+            return $values;
+        }
 
-                    default:
-                        break;
-                }
+        $referralCode = explode(',', $referralCode);
 
-                // Is this a referral?
-                if (strtolower($value[0]) === 'referral' && strtolower($value[1]) === 'true') {
-                    $referral = true;
-                }
+        foreach ($referralCode as $value) {
+            // See if we are dealing with ":" or "="
+            if (strpos($value, ':')) {
+                $value = explode(':', $value);
             }
-        }
-
-        // See if we have all the required information we need
-        if (!array_has($values, ['northstar_id', 'campaign_id', 'campaign_run_id'])) {
-            // If we have valid campaign values, use em! This also means that we do not have NS id
-            if (array_has($values, ['campaign_id', 'campaign_run_id']) && is_numeric($values['campaign_id']) && is_numeric($values['campaign_run_id'])) {
-                $finalValues = [
-                    'northstar_id' => null, // set the user to null so we force account creation when the code is not present.
-                    'campaign_id' => $values['campaign_id'],
-                    'campaign_run_id' => $values['campaign_run_id'],
-                    'source' => $this->source,
-                    'source_details' => null,
-                ];
+            elseif (strpos($value, '=')) {
+                $value = explode('=', $value);
             }
-
-            // If we have NS id, use it! This also means that we do not have both campaign_id and campaign_run_id, so use the defaults
-            if (array_has($values, ['northstar_id'])) {
-                $finalValues = [
-                    'northstar_id' => $values['northstar_id'], // set the user to null so we force account creation when the code is not present.
-                    'campaign_id' => 8017,
-                    'campaign_run_id' => 8022,
-                    'source' => $this->source,
-                    'source_details' => null,
-                ];
+            $key = strtolower($value[0]);
+            if ($key === 'user') {
+                $values['user_id'] = $value[1];
+            }
+            if (($key === 'campaignid' || $key === 'campaign') && is_numeric($value[1])) {
+                $values['campaign_id'] = (int) $value[1];
```

Comment:
Is there a need currently to send campaign ID from a referral string? Shouldn't this just be a server-side config variable per https://dosomething.slack.com/archives/CA0N5FXND/p1551114468011700?thread_ts=1551114281.011600&cid=CA0N5FXND